core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -0,0 +1,131 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.Nullable;
+
+import com.google.android.exoplayer2.Player;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.ads.AdPlaybackState;
+import com.google.android.exoplayer2.source.ads.AdsLoader;
+import com.novoda.noplayer.AdvertBreak;
+import com.novoda.noplayer.AdvertsLoader;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class NoPlayerAdsLoader implements AdsLoader, Player.EventListener {
+
+    private final AdvertsLoader loader;
+    @Nullable
+    private Player player;
+    @Nullable
+    private AdPlaybackState adPlaybackState;
+    @Nullable
+    private EventListener eventListener;
+
+    private int adIndexInGroup = -1;
+    private int adGroupIndex = -1;
+
+    NoPlayerAdsLoader(AdvertsLoader loader) {
+        this.loader = loader;
+    }
+
+    @Override
+    public void setSupportedContentTypes(int... contentTypes) {
+        // no-op
+    }
+
+    @Override
+    public void start(final EventListener eventListener, AdViewProvider adViewProvider) {
+        this.eventListener = eventListener;
+        if (adPlaybackState == null) {
+            loader.load(advertsLoadedCallback);
+        }
+    }
+
+    private final AdvertsLoader.Callback advertsLoadedCallback = new AdvertsLoader.Callback() {
+        @Override
+        public void onAdvertsLoaded(List<AdvertBreak> advertBreaks) {
+            adPlaybackState = AdvertPlaybackState.from(advertBreaks);
+
+            new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
do we need to create a new Handler every single time?

Reply:
Probably not :+1: